presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -279,7 +279,7 @@ private Date getDate(int columnIndex, DateTimeZone localTimeZone)
             return new Date(calendar.getTimeInMillis());
         }
         catch (IllegalArgumentException e) {
-            throw new SQLException("Invalid date from server: " + value, e);
+            throw new IllegalArgumentException("Expected column to be a date type but is " + columnInfo(columnIndex).getColumnTypeName());
```

Comment:
if our ability to convert to Date is dependent on the _value_ (rather than _type_), the exception message should quote the actual value (and the type is not relevant, so can be omitted)